docs/rules/no-render-return-undefined.md

```diff
@@ -0,0 +1,62 @@
+# Disallow returning undefined from react components (`react/no-render-return-undefined`)
+
+💼 This rule is enabled in the ☑️ `recommended` [config](https://github.com/jsx-eslint/eslint-plugin-react/#shareable-configs).
+
+<!-- end auto-generated rule header -->
+
+> In React 18, components may render undefined, and React will render nothing to the DOM instead of throwing an error. However, accidentally rendering nothing in a component could still cause surprises. This rule will warn if the `return` statement in a React Component returns undefined.
+
+Issue: [#3020](https://github.com/jsx-eslint/eslint-plugin-react/issues/3020)
+
+## Rule Details
+
+This rule will warn if the `return` statement in a React component returns undefined.
+
+Examples of **incorrect** code for this rule:
+
+```jsx
+function App() {
+  return undefined;
+}
+
+// OR
+
+let ui;
+function App() {
+  return ui;
+}
+
+// OR
+
+function getUI() {
+  if (condition) return <h1>Hello</h1>;
+}
+function App() {
+  return getUI();
+}
```

Comment:
this won't be possible across files, only if it's in the same file, so unless the function provides return type information (ie, TS) i don't think this is a check we can reliably do.

Reply:
Is it possible to continue without supporting this?

Reply:
@ljharb Can we do something here? 
Rest all changes, I've made

Reply:
Yes, I don't think we necessarily have to support this - I'm saying that the docs should be updated to reflect the caveats.

Reply:
Yes, I don't think we necessarily have to support this - I'm saying that the docs should be updated to reflect the caveats.